src/coreclr/jit/importer.cpp

```diff
@@ -21146,58 +21146,70 @@ void Compiler::impDevirtualizeCall(GenTreeCall*            call,
                     //
                     if (requiresInstMethodTableArg)
                     {
-                        const DWORD derivedClassAttribs = info.compCompHnd->getClassAttribs(derivedClass);
-
-                        if ((derivedClassAttribs & CORINFO_FLG_SHAREDINST) != 0)
+                        // Defer if prejitting as working backwards from the class handle to the right
+                        // representation is messy.
+                        if (opts.jitFlags->IsSet(JitFlags::JIT_FLAG_PREJIT))
                         {
-                            JITDUMP(
-                                "unboxed entry needs MT arg, but the handle we have is shared. Deferring update.\n");
+                            JITDUMP("unboxed entry needs MT arg, embedding this is complicated for prejit. Deferring "
```

Comment:
I believe that we have done work to embed this in crossgen2. Is this just a problem for old crossgen?

We should just throw an NYI exception in old crossgen instead of asserting. It is what we have been doing in other similar situations. We are generally not spending time on keeping parity in old crossgen.

Reply:
Yes, it's just a problem for crossgen, and only for a relatively small number of methods.

So I'll revise per your suggestion.

Reply:
Crossgen is already throwing here -- but it is asserting first. So looks like I can just remove the assert.

https://github.com/dotnet/runtime/blob/98ac23212e6017c615e7e855e676fc43c8e44cb8/src/coreclr/zap/zapinfo.cpp#L1578-L1582